src/components/general/DatePicker/index.tsx

```diff
@@ -33,25 +40,54 @@ const DatePicker: React.FC<DatePickerProps> = ({ disabled, error, errorMessage,
             return formatDate(selectedDate);
         }, [selectedDate, isOpen, maskedValue]);
 
-        const handleKeyUp = useCallback((e: React.KeyboardEvent<HTMLInputElement>) => {
-            if(e.keyCode === 13 && isValidMask) {
-                try {
-                    const parsedDate = parseDate(maskedValue);
-                    onChange(parsedDate);
+        const tryParseDate = useCallback(() => {
+            try {
+                const parsedDate = parseDate(maskedValue);
+                onChange(parsedDate);
+            } catch (e) {}
+        }, [maskedValue, parseDate]);
+
+        const handleKeyUp = useCallback(

```

Comment:
when only using codes, please provide in comment which key has been use.
tip: import lib for mapping name => keycode

Reply:
Replace deprecated `KeyboardEvent.keyCode` with `KeyboardEvent.key`